src/rise-lisk/dposutils.c

```diff
@@ -103,7 +103,7 @@ uint64_t deriveAddressFromPublic(cx_ecfp_public_key_t *publicKey) {
 uint32_t setSignContext(commPacket_t *packet) {
   // reset current result
   uint8_t tmp[256];
-  os_memset(&signContext.digest, 0, 32);
+  os_memset(signContext.digest, 0, 32);
```

Comment:
hmm?

Reply:
wondering how it was properly working before.

Reply:
Yeah, that had me scratching my head as well. Here's two possible explanations why it worked before:

1) In some cases the pointer of a array-pointer ends up pointing to the same memory location. Since memset doesn't really care about the pointer type, it ends up working.

2) Both the read and write used the pointer of the array-pointer, thus the actual data ended up in some random memory location instead of the array allocated to it.

Most likely what we had was 1. Even though it wasn't correct, it ended up working in that situation (got lucky basically).